test/tests/functional/pbs_pbsnodes_attributes.py

```diff
@@ -0,0 +1,115 @@
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class Testpbsnodes_attributes(TestFunctional):
+
+    def Test_pbsnodes(self, user):
+        """
+        Function to validate pbsnodes attribute on user basis
+        """
+
+        attr_dict = {}
+        host_short_name = self.mom.shortname
+        host_name = self.mom.hostname
+        pbs_version = self.server.pbs_version
+
+        command = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                               'bin', 'pbsnodes -a')
+        if user == 'root':
+            ret = self.du.run_cmd(self.server.hostname, command, sudo=True)
+        else:
+            ret = self.du.run_cmd(self.server.hostname, command, runas=user)
+
+        #converting pbsnodes -a output into dictionary for validation
+        attr_list = ret['out']
+        list_len = len(attr_list) - 1
+        attr_list = filter(None, attr_list)
+        for i in range(1, list_len):
+            attr = attr_list[i].split('=')[0].strip()
+            val = attr_list[i].split('=')[1].strip()
+            attr_dict[attr] = val
+
+        if 'Mom' in attr_dict:
+            self.assertEqual(attr_dict['Mom'], host_name)
+        if 'ntype' in attr_dict:
+            self.assertEqual(attr_dict['ntype'], 'PBS')
+        if 'state' in attr_dict:
+            self.assertEqual(attr_dict[ATTR_NODE_state], 'free')
+        if 'resources_available.arch' in attr_dict:
+            self.assertEqual(attr_dict['resources_available.arch'], 'linux')
+        if 'resources_available.vnode' in attr_dict:
+            self.assertEqual(attr_dict['resources_available.vnode'],
+                             host_short_name)
+        if 'resources_available.host' in attr_dict:
+            self.assertEqual(attr_dict['resources_available.host'],
```

Comment:
where is resources_available.ncpus and pcpus, resources_available.mem?
Also I see 'last_state_change_time' set to date. I suggest just verifying it is set to date format is sufficient.